### DISCUSSION TITLE
Documentation about autocomplete

### DISCUSSION THREAD

**Discussion by hembo** [directus/directus#7566] [Archive]
Hello, I wonder if there are any documentation about how to use autocomplete field type?
I have set up the api uri but the value for the query is not sent to api.
My uri is http://viaf.org/viaf/AutoSuggest?query=
Result path is: result.term
And Value path is: result

Also, as the api gives back more that one field, i wonder how i could store the text value as well as numeric reference?

Many thanks in advance
Hembo

**Comment by rijkvanzanten** [comment]
> My uri is http://viaf.org/viaf/AutoSuggest?query=

You'll have to tell it where to put the value that you enter in the interface:

`http://viaf.org/viaf/AutoSuggest?query={{ value }}`

**Reply by rijkvanzanten** [reply]
> Also, as the api gives back more that one field, i wonder how i could store the text value as well as numeric reference?

That's a very good question. Right now, it only saves the path as stored in `valuePath` 🤔 I'm wondering if we figure out a way to save two values into the same field (maybe using another template), or have a way to fetch a single request and have it stored in more than one field (though that might be a bit trickier to realize)

**Reply by simgero** [reply]
@rijkvanzanten  maybe it is possible to implement the autocomplete feature for the field type "JSON" and then save the whole result? Then the appearance can be changed via the display settings.

**Reply by rijkvanzanten** [reply]
Makes sense to me! Wanna open a PR for that?

**Reply by hembo** [reply]
that would be awesome!

**Reply by hembo** [reply]
Also looking now my datamodel again. https://viaf.org/viaf/AutoSuggest?query=Leonardo gives me back json whee i have results in result.term field. When i used that (result.term) for Results Path and Value Path no values where displayed.

**Reply by rickysullivan** [reply]
I've run into the same issue and created a custom interface (forked from the built-in one) for it: https://github.com/directus/directus/discussions/10128#discussioncomment-4649071
